services/dbaas/redisdb/replica_set.go

```diff
@@ -155,7 +156,13 @@ func (c *RedisDBClient) SetRedisDBReplicaSetData(d *schema.ResourceData, replica
 		}
 	}
 
-	// TODO -- Check if we need to set anything for 'credentials'
+	if replicaSet.Properties.Credentials != nil {
```

Comment:
@avirtopeanu-ionos for `replicaSet.Properties.MaintenanceWindow` we have a different approach that looks cleaner, but you achieve the same thing using one line, so I think it's better to modify all the sets to use this format: `[]interface{}{setFunction(*replicaSet.Properties.Field)}`, please add a `TODO` for this.

Reply:
can we do this in the main branch?

Reply:
@avirtopeanu-ionos all the `TODO`s are just some references for our PR. Before merging, no `TODO`s will be left.